arm-containerregistry/2017-06-01-preview/swagger/containerregistry.json

```diff
@@ -0,0 +1,1744 @@
+          "description": "The SKU of the container registry."
+        },
+        "properties": {
+          "$ref": "#/definitions/RegistryProperties",
+          "description": "The properties of the container registry.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "Sku": {
+      "description": "The SKU of a container registry.",
+      "required": [
+        "name"
+      ],
+      "properties": {
+        "name": {
+          "description": "The SKU name of the the container registry. Required for registry creation.",
+          "enum": [
+            "Basic",
+            "Standard",
+            "Premium"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "SkuName",
+            "modelAsString": true
+          }
+        },
+        "tier": {
+          "description": "The SKU tier based on the SKU name.",
+          "enum": [
+            "Basic",
+            "Standard",
+            "Premium"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "SkuTier",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "RegistryProperties": {
+      "description": "The properties of a container registry.",
+      "properties": {
+        "loginServer": {
+          "description": "The URL that can be used to log into the container registry.",
+          "type": "string",
+          "readOnly": true
+        },
+        "creationDate": {
+          "format": "date-time",
+          "description": "The creation date of the container registry in ISO8601 format.",
+          "type": "string",
+          "readOnly": true
+        },
+        "provisioningState": {
+          "description": "The status of the container registry at the time the operation was called.",
+          "enum": [
+            "Creating",
+            "Succeeded"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "ProvisioningState",
+            "modelAsString": false
+          }
+        },
+        "adminUserEnabled": {
+          "description": "The value that indicates whether the admin user is enabled.",
+          "default": false,
+          "type": "boolean"
+        },
+        "storageAccount": {
+          "$ref": "#/definitions/StorageAccountProperties",
+          "description": "The properties of the storage account for the container registry. Only applicable for Basic SKU."
+        }
+      }
+    },
+    "StorageAccountProperties": {
+      "description": "The properties of a storage account for a container registry. Only applicable for Basic SKU.",
+      "properties": {
+        "name": {
+          "description": "The name of the storage account.",
+          "type": "string"
+        }
+      }
+    },
+    "RegistryCreateParameters": {
+      "description": "The parameters for creating a container registry.",
```

Comment:
I made the change but they are technically not resources (there is no such a resource called `RegistryCreateParameters`) but only parameters to create resources. And this change doesn't affect the payload either. It is now complaining that we should provide get/list/etc. operations for `RegistryCreateParameters` and `WebhookCreateParameters` because they are tracked resources, which shouldn't be. We have been following [this pattern from storage account](https://github.com/Azure/azure-rest-api-specs/blob/master/arm-storage/2016-12-01/swagger/storage.json#L665) and using it for a while. What is the exact benefit of enabling those validation rules and making this change? I am leaning towards reverting the change now, but please let me know.